linkml/generators/pydanticgen/template.py

```diff
@@ -195,11 +195,13 @@ class PydanticAttribute(PydanticTemplateModel):
     def field(self) -> str:
         """Computed value to use inside of the generated Field"""
         if self.predefined:
-            return self.predefined
+            value = self.predefined
         elif self.required or self.identifier or self.key:
-            return "..."
+            value = "..."
         else:
-            return "None"
+            value = "None"
+
+        return f"default={value}"
```

Comment:
I would prefer that this happens in the template, rather than the template model - the template model is intended as computer-readable data that can be serialized using a template, where all the information about the representation as a concrete pydantic module is in the templates, and the template models are just data containers. In general I think the amount of string manipulation that happens in pydanticgen and other generators makes them fragile and opaque, so it's my preference that we keep it at a minimum :). Promise i am not just being nitpicky.

See how the rest of the kwargs are specified in the template here: https://github.com/linkml/linkml/blob/08959cb73d8b09d66bbad730bb5f4d63f0195b08/linkml/generators/pydanticgen/templates/attribute.py.jinja#L1

Reply:
Oh good catch! I’m still getting used to the template model, but I agree about keeping the value as a pure value and formatting in the Jinja.

Reply:
one of those "seems obvious in retrospect but the value of modeling both the source and the target domain was only made clear after a lot of pain and wasted time (mostly pre-linkml)" lol